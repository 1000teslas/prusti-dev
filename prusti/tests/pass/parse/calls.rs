// compile-flags: -Zprint-desugared-specs -Zprint-typeckd-specs -Zhide-uuids
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"

#![feature(register_tool)]
#![register_tool(prusti)]

use prusti_contracts::*;


#[ensures(result == a || result == b)]
#[ensures(result >= a && result >= b)]
fn max(a: i32, b: i32) -> i32 {
    if a > b {
        a
    } else {
        b
    }
}

fn test_max1() {
    let a = 5;
    let b = 6;
    let z = max(a, b);
    assert!(z == 6);
}

fn test_max2() {
    let a = 5;
    let b = 6;
    let z = max(a, b);
    assert!(z == 5);
}

fn main() {}
