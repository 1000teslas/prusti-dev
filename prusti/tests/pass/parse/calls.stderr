Verification of 5 items...
error: [Prusti: verification error] statement might panic
  --> $DIR/calls.rs:28:5
   |
28 |     assert!(z == 5);
   |     ^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/calls.rs:35:9
   |
35 |         result == 3     // test that we get correct span information
   |         ^^^^^^^^^^^
   |
note: the error originates here
  --> $DIR/calls.rs:43:1
   |
43 | / fn test_max3() -> i32 {
44 | |     let a = 4;
45 | |     let b = 3;
46 | |     max(a, b)
47 | | }
   | |_^

Verification failed
error: aborting due to 2 previous errors

